lib/services/traits.go

```diff
@@ -108,9 +137,27 @@ func traitsToRoles(ms types.TraitMappingSet, traits map[string][]string, collect
 					}
 				}
 			}
+
+			// warn at most maxMismatchedTraitValuesWarned trait values to prevent huge warning lines
+			if len(mismatched) > maxMismatchedTraitValuesWarned {
+				warnings = append(warnings, fmt.Sprintf(
```

Comment:
I know I suggested adding this to the returned warnings myself but now that I'm thinking about it again I wonder if we'll only be increasing amount of log noise this way - because these returned warnings are logged at "warn" level on the caller's side. I think we should just log these "mismatched" traits locally in this method and keep it at debug.

Reply:
I had similar thoughts here. +1.

Reply:
Done in 15df5389a3dc66a3785349e20d4a1bb88ca3e22b.